src/jmh/java/column/store/input/InMemoryReader.java

```diff
@@ -0,0 +1,274 @@
+package column.store.input;
+
+import column.store.api.column.*;
+import column.store.api.query.Query;
+import column.store.api.read.*;
+
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.nio.ByteBuffer;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.NoSuchElementException;
+
+/**
+ * Reads all records into memory column by column.
+ */
+public class InMemoryReader implements Reader {
+
+    private final Map<Column, ByteReader> readers;
+    private Path root;
+
+    private boolean consumed = true;
+    private boolean hasNext;
+
+    public InMemoryReader() {
+        this.readers = new HashMap<>();
+    }
+
+    @Override
+    public void query(final Query query) {
+        if (!query.filters().isEmpty()) {
+            throw new UnsupportedOperationException("For performance reasons, this reader does not support filters");
+        }
+        readers.clear();
+        root = query.filePath();
+    }
+
+    @Override
+    public BooleanByteReader of(final BooleanColumn column) {
+        return (BooleanByteReader) readers.computeIfAbsent(column, col -> {
+            var bytes = readAllBytesOf(column);
+            return new BooleanByteReader(bytes);
+        });
+    }
+
+    @Override
+    public DoubleByteReader of(final DoubleColumn column) {
+        return (DoubleByteReader) readers.computeIfAbsent(column, col -> {
+            var bytes = readAllBytesOf(column);
+            return new DoubleByteReader(bytes);
+        });
+    }
+
+    @Override
+    public IdByteReader of(final IdColumn column) {
+        return (IdByteReader) readers.computeIfAbsent(column, col -> {
+            var bytes = readAllBytesOf(column);
+            return new IdByteReader(bytes);
+        });
+    }
+
+    @Override
+    public LongByteReader of(final LongColumn column) {
+        return (LongByteReader) readers.computeIfAbsent(column, col -> {
+            var bytes = readAllBytesOf(column);
+            return new LongByteReader(bytes);
+        });
+    }
+
+    @Override
+    public StringByteReader of(final StringColumn column) {
+        return (StringByteReader) readers.computeIfAbsent(column, col -> {
+            var bytes = readAllBytesOf(column);
+            return new StringByteReader(bytes);
+        });
+    }
+
+    private byte[] readAllBytesOf(final Column column) {
```

Comment:
attempt to read all Bytes into memory, could result in a OutOfMemory Exception.
-> large number of records + many colums

Reply:
Yes, but but how big will our test set realistically be? If it is too big, we can always switch to something more sophisticated.